src/main/java/fruit/shop/strategy/ActivityStrategyImpl.java

```diff
@@ -0,0 +1,44 @@
+package fruit.shop.strategy;
+
+public class ActivityStrategyImpl implements ActivityStrategy {
+    @Override
+    public int handleTransaction(String strOption, int currentValue, int value) {
+        OptionRepresentation option = OptionRepresentation.getOptionRepresentation(strOption);
+        switch (option) {
+            case BALANCE:
+                return new BalanceHandler().getOptionResult(currentValue, value);
+            case SUPPLY:
+                return new SupplyHandler().getOptionResult(currentValue, value);
+            case PURCHASE:
+                return new PurchaseHandler().getOptionResult(currentValue, value);
+            case RETURN:
+                return new ReturnHandler().getOptionResult(currentValue, value);
+            default:
+                throw new RuntimeException("There is no such strOption!");
+        }
+    }
+
+    private enum OptionRepresentation {

```

Comment:
This Enum should as part of model FruitTransaction or independent part
use the example from instruction
```suggestion
    private enum Operation {
```